cliclient/cliclient.go

```diff
@@ -99,8 +126,31 @@ func NewProjectClient(config *config.ServerConfig) (*MasterClient, error) {
 	return mc, nil
 }
 
+var testingForceClientInsecure = false
+
+/*
+TestingForceClientInsecure sets testForceClientInsecure to true for unit tests.
+It's a workaround to github.com/rancher/norman/clientbase.NewAPIClient,
+which replaces net/http.Client.Transport (including proxy and TLS config),
+so the client TLS config of net/http/httptest.Server will be lost.
+Not thread-safe.
+Call the returned function by defer keyword, for example:
+
+	defer cliclient.TestingForceClientInsecure()()
+*/
+func TestingForceClientInsecure() func() {
+	origTestForceClientInsecure := testingForceClientInsecure
+	testingForceClientInsecure = true
+	return func() {
+		testingForceClientInsecure = origTestForceClientInsecure
+	}
+}
+
 func (mc *MasterClient) newManagementClient() error {
 	options := createClientOpts(mc.UserConfig)
+	if testingForceClientInsecure {
```

Comment:
Personally, I would avoid adding conditions for tests into main code.

Reply:
Me too, but the config.ServerConfig does not support the insecure flag.

If you mind, I can extend the config.ServerConfig with the insecure flag, annotated with `json:"-"`, to make sure, it won't be loaded from config file.

Reply:
Can you explain why we need to set the `Insecure` for tests?

Reply:
A httptest.Server was started, which supports TLS (client TLS config can be get), but it was unable to use, because github.com/rancher/norman/clientbase.NewAPIClient replaces net/http.Client.Transport (including proxy and TLS config),
+so the client TLS config of net/http/httptest.Server will be lost. I cannot understand why NewAPIClient  was designed this way.

So insecure must be used with httptest.Server.